### PR TITLE
Destination Snowflake: Return Standard Loading

### LINKED ISSUE #4370: Snowflake destination: Support internal staging
## Tell us about the problem you're trying to solve
Records can be inserted into snowflake using a standard SQL `INSERT` statement or by first copying the data to blob storage (e.g: azure, gcs, s3, snowflake internal) then using the `COPY` command to read that data into snowflake. The staging approach is an order of magnitude faster and should pretty much always be used except for the smallest data volumes/proof of concepts. 

The snowflake destination currently supports staging to S3. it should also support internal staging as described here: https://docs.snowflake.com/en/user-guide/data-load-considerations-stage.html

## Describe the solution you’d like
Support internal staging in snowflake. 

## Acceptance criteria
1. Internal staging is supported
2. All docs are updated with explicit documentation of needed DB user permissions
3.  If no backwards incompatible changes are introduced in terms of permissions, remove the INSERT mode in a backwards compatible way. If we have internal staging then there is basically no reason to use inserts

### PR DESCRIPTION
**Pull request by VitaliiMaltsev**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json [hunk 1]
```diff
@@ -87,6 +87,19 @@
               }
             }
           },
+          {
+            "title": "Standard Inserts",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json:91]: can you: 

1. set `hidden: true` to hide this option from the UI (no one should be using it anymore)
2. in the connector, use internal staging when this option is set

Comment (tuliren) [reply airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json:91]: @sherifnada, when `hidden` is set to `true`, what will the UI show for a connection already set to standard loading? Will it cause a UI exception?

**Review by tuliren — APPROVED** [review]

File: airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json (hunk 1 above)
Comment (sherifnada) [reply airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json:91]: actually nvm, let's just do this option for now and handle what I suggested in a follow up PR just to unblock current users

Comment (sherifnada) [reply airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json:91]: I think `hidden` won't work for enum values actually so I was wrong. migration will be a little more complex

**Comment by tuliren** [comment]
@VitaliiMaltsev, FYI, sorry that we actually should remove `hidden`. See the discussion in the comments above. Could you update it before publishing? Also don't forget to bump the version, and update the CHANGELOG.

File: airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json (hunk 1 above)
Comment (VitaliiMaltsev) [reply airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json:91]: Yes, hidden: true won't work for oneOf. Just tested it on UI

**Comment by tuliren** [comment]
@VitaliiMaltsev, yes, feel free to publish the new version. Thanks~~

**Pull request merged**